### DISCUSSION TITLE
Week 12 Project Polishing

### DISCUSSION THREAD

**Discussion by jakegros** [USU-ITLS-x265-23-summer/class-repo#228] [General]
I need a lot of polish for my project.  It has come a long way for me, though.  
I can't seem to get my dropdown menus to go away when I click on another menu, only when clicked outside of them. 
Some items accomplished this week:
- Added footer
- Built up a lot of content on each page 
- Added a quote and formatting 
- Completed responsive Conatact form 
- Added more styling and JS

I still need to:
- Hover links...
- Add ids 
- Fix dropdown menus, link to pages
- Semantic html
- Perhaps have quote run down right side parallel to content info?

Let me know what thoughts you have for my site.
https://github.com/jakegros/project
https://jakegros.github.io/project/

**Comment by beto0890** [comment]
Great website! all the buttons are working well and I like the quote in the middle with the blue background to emphasize it.

**Reply by jakegros** [reply]
Thank you!  I appreciate it.  What do you think it is missing or could use?

**Reply by CJ3315** [reply]
The hover links are going to be a great addition! Let me know if you need any help with those!

**Reply by bst3ck** [reply]
Love it as well. I like the quote block too. Looks great ;)
I think the flow works really well. I liked the expansion buttons, but it may be good to have a function that hides the open drop down menus when you open another drop down. But it looks great.

**Comment by derekscholes** [comment]
Here is some constructive feedback I have:
- Buttons on the top of the page can use some CSS as they are small and hard to read on big screens. 
- There is an extra bullet on Closed Captioning
- I like larger margins and font, but that is a personal preference

Otherwise, everything looks good!

**Reply by jakegros** [reply]
Thank you for the feedback, everyone. I appreciate your thoughtfulness and testing it out.  I may need help getting the drop down menus to close when another opens. 🙏

**Reply by bst3ck** [reply]
I would probably write a new function that I would add to the functions you already have the opens them up. Before you trigger to show option, loop through the elements and close them all. Maybe something like this:

```
function closeAllDropdowns() {
    
    const collection = document.getElementsByClassName("dropdown-content");
    for (var i = 0; i < collection.length; i++) {
        collection[i].classList.remove("show")
    }

}
```

Of course this was not tested, but it gives a high level idea.

**Comment by audreyhki** [comment]
Nice job! The site is looking really nice, I clicked around and went through all of the pages up top and had no issues with any of them working.

**Comment by gabebott** [comment]
Great job so far! I think you are just about there on it. I think it is just missing those steps that you mentioned and then I have no doubt that it will be all ready for grading. Great job Jake!

**Comment by kassyoveson** [comment]
I like the look of your page. I agree with Derek, that the font needs to be a little bit bigger to see it well on big screens. All of your links worked which is a great accomplishment. Great job!
